### PR TITLE
bash script to call python script, minor pipeline and README changes

### PR DESCRIPTION
**Pull request by avinashtrivedi11**
- created bash script which calls the python script and gets reverse mapping 
- including python and bash script in central repo pipeline from remote repo using curl
- updated central README with knowledge portal changes scenarios
- source pipeline fix to not use 'Detecttechnologies/' to split repo path
-  no need to use 'my/prefix' in docs-manifest
- other minor changes

### CODE REVIEW

File: knowledge-portal/central/reverse_mapping.py [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+# manifests/<repo_name>.txt: Source repo name should be the manifest name
+# manifests have the convention: `source-->destination`
+import os
+import subprocess
+import sys
+import time
+from glob import glob
+
+
+def sp_run(s):
```
Comment (ashwinr64) [inline knowledge-portal/central/reverse_mapping.py:12]: You can remove this aswell as the commented lines in the end

Comment (avinashtrivedi11) [reply knowledge-portal/central/reverse_mapping.py:12]: Ohh yeah sorry, will do that

**Review by ashwinr64 — APPROVED** [review]
LGTM. @rsnk96 pls merge once you look at it.

File: knowledge-portal/central/initiate_push.sh [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+# changed files can be obtained by git diff between latest commit and unstaged area
+changed_files=$(git diff --name-only HEAD);
+echo "Files changed in knowledge portal: ${changed_files}";
+
+if [[ $changed_files ]]
+then
+    # Reverse the git reset and move files to staged as we will need latest files 
+    git add -A && git commit -m "docs:reversing soft reset as now we have the filenames"
+
+    # Run python script to get output in format "source_repo_name:rhs_path-->lhs_path"
```
[the remark this answers is not in the document]
Comment (avinashtrivedi11) [reply knowledge-portal/central/initiate_push.sh]: Done

File: knowledge-portal/central/README.md [hunk 3]
```diff
@@ -14,4 +14,27 @@ include:
 1. Go to `Settings >> Access Tokens`.
 2. Set `Token name = PUSH_TOKEN`. This is important and you can't choose any other name without modifying the source pipeline as well.
 3. Select the role `Maintainer` and `write_repository` as the scope of token.
-4. `Create project access token` will create a unique token for you that can then be shared with people setting up the `knowledge-portal/source` pipeline.
+4. `Create project access token` will create a unique token for you that can then be shared with people setting up people setting up individual `source` pipelines.
```
[the remark this answers is not in the document]
Comment (avinashtrivedi11) [reply knowledge-portal/central/README.md]: This was only meant as a 'mistake'.

File: knowledge-portal/central/initiate_push.sh [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+        if [ -d "$unique_directory" ]
+        then
+            # Remove existing and Copy changed files to unique_directory
+            rm -rfv ${unique_directory}/${lhs_path}
+            cp -R -v ./"${rhs_path}" "${unique_directory}/${lhs_path}"
+        else
+            # Create directory, clone source repo, create branch, and initiate copy
+            mkdir -p ${unique_directory}
+            source_repo_array+=( "${source_repo_name}" )
+            git clone -v https://oauth2:${BOT_ACCESS_TOKEN}@gitlab.com/DetectTechnologies/${source_repo_path}.git ${unique_directory}
```
Comment (rsnk96) [inline knowledge-portal/central/initiate_push.sh:51]: Can we document in the readme that this `BOT_ACCESS_TOKEN` should also be defined as a variable by the maintainers centrally in the org-wide gitlab instance settings?

Comment (avinashtrivedi11) [reply knowledge-portal/central/initiate_push.sh:43]: Done

**Pull request merged**